libs/domain/site/src/services/breadcrumbs/default-breadcrumbs.service.ts

```diff
@@ -0,0 +1,55 @@
+import { INJECTOR, inject } from '@spryker-oryx/di';
+import { RouterService } from '@spryker-oryx/router';
+import { Observable, map, switchMap } from 'rxjs';
+import { Breadcrumb } from '../../models';
+import {
+  BreadcrumbsResolver,
+  BreadcrumbsResolvers,
+  BreadcrumbsService,
+} from './breadcrumbs.service';
+
+export class DefaultBreadcrumbsService implements BreadcrumbsService {
+  constructor(protected injector = inject(INJECTOR)) {}
+
+  protected homeBreadcrumb = {
+    i18n: { token: 'breadcrumbs.home' },
+    url: '/',
+  };
+
+  protected getResolverKey(type: string): string {
+    return `${BreadcrumbsResolvers}${type}`;
+  }
+
+  protected resolveType(): Observable<string | undefined> {
+    return this.injector
+      .inject(RouterService)
+      .current()
+      .pipe(map((route) => route.type as string | undefined));
```

Comment:
do we need typecasting here?

Reply:
route type has special type `RouteType` those values are strings. Unfortunately we need to cast the type